src/payment/models/payout/strategies/payout-token.strategy.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from '@nestjs/common';
+import { NodeService } from 'src/ain/node/node.service';
+import { MailService } from 'src/shared/services/mail.service';
+import { DexService } from '../../dex/services/dex.service';
+import { PayoutOrder, PayoutOrderContext } from '../entities/payout-order.entity';
+import { PayoutOrderRepository } from '../repositories/payout-order.repository';
+import { PayoutDeFiChainService } from '../services/payout-defichain.service';
+import { PayoutStrategy } from './payout.strategy';
+
+type TokenName = string;
+
+@Injectable()
+export class PayoutTokenStrategy extends PayoutStrategy {
```

Comment:
Same question as last time 😉 I really like the strategy thinking ... but again there is a lot of duplicated code. E.g. the `sendToken` and `sendDFI` methods are practically the same. And it's not some trivial code but a very fundamental logic.
Is there a way to combine it? 😊

Reply:
hmm, was trying to do that already and put everything shared in superclass, but totally open to suggestions, lets discuss this